report/topology.go

```diff
@@ -184,12 +184,30 @@ func (t Topology) EdgeMetadata(mapFunc MapFunc, srcRenderableID, dstRenderableID
 
 // Squash squashes all non-local nodes in the topology to a super-node called
 // the Internet.
+// We rely on the values in the t.Adjacency lists being valid keys in
+// t.NodeMetadata (or t.Adjacency).
 func (t Topology) Squash(f IDAddresser, localNets []*net.IPNet) Topology {
-	isRemote := func(ip net.IP) bool { return !netsContain(localNets, ip) }
+	isRemote := func(id string) bool {
+		_, ok := t.NodeMetadatas[id]
+		if ok {
+			return false
+		}
+
+		// TODO make MakeAdjacencyID just take a single argument,
+		// and ParseAdjacencyID do the right thing.
```

Comment:
Let's make this change now rather than leaving it as a TODO. I'm OK with changing the function definition in id.go as long as we leave the warning comments.